src/profile-logic/profile-data.js

```diff
@@ -158,34 +158,19 @@ export function getFuncIndex(path: CallNodePath): IndexIntoFuncTable {
   return path[path.length - 1];
 }
 
-/**
- * This function Returns the JS implementation information for a specific stack.
- */
-export function getJsImplementationForStack(
-  stackIndex: IndexIntoStackTable,
-  { stackTable, frameTable, stringTable }: Thread
-): string {
-  const frameIndex = stackTable.frame[stackIndex];
-  const jsImplementationStrIndex = frameTable.implementation[frameIndex];
-  const jsImplementation =
-    jsImplementationStrIndex === null
-      ? 'interpreter'
-      : stringTable.getString(jsImplementationStrIndex);
-
-  return jsImplementation;
-}
-
-type Breakdown = { [key: string]: Milliseconds } | null;
+type JsImplementation = 'interpreter' | 'ion' | 'baseline' | 'unknown';
```

Comment:
I'm curious, do we ever have unknown values, or are you being cautious here? Should we fail or do a `console.error` for this case?

Reply:
We don't have any at the moment, but I think we may have new implementations in the future, and if this happens, the JS engine developers might forget telling us. That's why I didn't want to fail too badly. I could add a console.error. Maybe in the future we should have an error log somewhere in the UI.

Also I don't really know what happens with asm.js or wasm, so maybe we already have more values and we just don't know...